RedditSharp/Exceptions/RedditException.cs

```diff
@@ -8,6 +9,8 @@ namespace RedditSharp
     [Serializable]
     public class RedditException : Exception
     {
+        public JArray Errors { get; }
+
```

Comment:
Does this actually return the errors from Reddit properly? That's been something I've been meaning to fix up. It also needs an HTTP status code preferably.

Reply:
Yes, it does, but only for the new endpoints I've implemented. But yeah, I agree the code in `ExecuteRequestAsync` needs to be cleaned up to expose the errors instead of swallowing an exception.

Example response when trying to delete a collection that doesn't exist: `{"json": {"errors": [["INVALID_COLLECTION_ID", "That collection doesn't exist", "collection_id"]]}}` but the status code is `200`. 

"errors" is an array of arrays... kinda weird, but the consumer can do whatever with it

Reply:
@CrustyJew It appears RedditHttpException has the status code built in. Perhaps eventually having an `Error` class that can be used across the board?